src/client/Client.cc

```diff
@@ -12829,18 +12871,6 @@ size_t Client::_vxattrcb_client_id(Inode *in, char *val, size_t size)
 }
 
 const Client::VXattr Client::_dir_vxattrs[] = {
-  {
-    name: "ceph.dir.layout",
-    getxattr_cb: &Client::_vxattrcb_layout,
-    readonly: false,
-    exists_cb: &Client::_vxattrcb_layout_exists,
-    flags: 0,
-  },
-  XATTR_LAYOUT_FIELD(dir, layout, stripe_unit),
-  XATTR_LAYOUT_FIELD(dir, layout, stripe_count),
-  XATTR_LAYOUT_FIELD(dir, layout, object_size),
-  XATTR_LAYOUT_FIELD(dir, layout, pool),
-  XATTR_LAYOUT_FIELD(dir, layout, pool_namespace),
```

Comment:
Consider a new (Quincy) client talking to an old Ceph cluster (Pacific or older).

Reply:
@batrick The old method throws up ENODATA while fetching layout xattrs for dirs anyway.
Should I try and fix the original issue as well ?

Reply:
It only returns `ENODATA` if the layout is inherited or the default?

> Should I try and fix the original issue as well ?

What do you mean?

Reply:
> It only returns `ENODATA` if the layout is inherited or the default?
> 
> > Should I try and fix the original issue as well ?
> 
> What do you mean?

@batrick The current method to get vxattrs is broken and results in an `ENODATA` while getting individual layout vxattrs. This PR aims to fix this very problem. If we keep the old code, as per your recommendation, new clients (Quincy) will still not be able to get layout vxattr values, since the vxattr propagation mechanism is broken in existing code. Only if we fix the old code will clients be able to get vxattr values. But such a fix would anyway be available with newer code or backports fix to older releases.
I still haven't figured out how keeping the existing code for maintaing backward compatibility is going to help fix the issue ?
Or are you suggesting that I change the implementation to send a `getvxattr` RPC instead of relying on values in the `Inode` struct on the client side. But this is not going to work when new clients (Quincy and later) are interacting with older (Pacific and earlier) servers.
What am I missing here ?

Reply:
> Or are you suggesting that I change the implementation to send a getvxattr RPC instead of relying on values in the Inode struct on the client side. But this is not going to work when new clients (Quincy and later) are interacting with older (Pacific and earlier) servers.

Yes. If server responds with ENOSYS then fallback to using the Inode struct information.

You'll need to give me an example of ENODATA with layout vxattrs. TMK, the code works fine if the layout actually exists on the directory (not inherited or default) and on any file.

Reply:
@batrick thanks for the `ENOSYS` hint; please review the updated PR

Reply:
@batrick I tried testing latest client with nautilus server; the nautius server crashed! :P

Reply:
@batrick pacific server bails out with a `ceph_abort()` as well while trying to gather op latency for unknown `getvxattr` op for perf metrics

Reply:
client now checks if all mds ranks in the active set of sessions are at `quincy` before forwarding the request to the server